File: src/main/java/model/Budget.java
package model;

import java.util.ArrayList;
import java.util.List;

public class Budget {
    private double balance;
    private final List<Transaction> transactions;

    public Budget() {
        this.balance = 0.0;
        this.transactions = new ArrayList<>();
    }

    public void addTransaction(Transaction transaction) {
        if (transaction.getType() == TransactionType.INCOME) {
            balance += transaction.getAmount();
        } else {
            balance -= transaction.getAmount();
        }
        transactions.add(transaction);
    }

    public double getBalance() {
        return balance;
    }

    public List<Transaction> getTransactions() {
        return new ArrayList<>(transactions);
    }
}

File: src/models/Budget.js
export class Budget {
  constructor() {
    this.balance = 0.0;
    this.transactions = [];
  }

  addTransaction(transaction) {
    if (transaction.getType() === 'INCOME') {
      this.balance += transaction.getAmount();
    } else {
      this.balance -= transaction.getAmount();
    }
    this.transactions.push(transaction);
  }

  getBalance() {
    return this.balance;
  }

  getTransactions() {
    return [...this.transactions];
  }
}